src/os/bluestore/BlueStore.cc

```diff
@@ -11809,11 +11878,15 @@ void BlueStore::_read_cache(
   size_t length,
   int read_cache_policy,
   ready_regions_t& ready_regions,
-  blobs2read_t& blobs2read)
+  blobs2read_t& blobs2read,
+  span_stat_t* res_span_stat)
 {
   // build blob-wise list to of stuff read (that isn't cached)
   unsigned left = length;
   uint64_t pos = offset;
+  span_stat_t dummy_span_stat;
+  span_stat_t& span_stat = res_span_stat ? *res_span_stat : dummy_span_stat;
+  span_stat.stored = length;
   auto lp = o->extent_map.seek_lextent(offset);
   while (left > 0 && lp != o->extent_map.extent_map.end()) {
```

Comment:
Need to `span_stat.stored -= left;` after we exit the loop to factor in the case that we reached `end()` earlier.